pkgs/applications/altcoins/zcash/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, libsodium, fetchFromGitHub, wget, pkgconfig, autoreconfHook, openssl, db62, boost
+, zlib, gtest, gmock, miniupnpc, callPackage, gmp, qt4, utillinux, protobuf, qrencode, libevent
+, withGui }:
+
+let snark        = callPackage ./snark { inherit boost openssl; };
```

Comment:
Definitely (for aggregate computation over any private data that previously can't be achieved due to ethical reasons). I think it is more modular to package `libsnark` first to have less CR surface, though at the expense of having more roundtrips to `zcash` being merged.